packages/flutter/lib/src/services/text_editing_delta.dart

```diff
@@ -0,0 +1,400 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'text_editing.dart';
+import 'text_input.dart' show TextEditingValue;
+
+TextAffinity? _toTextAffinity(String? affinity) {
```

Comment:
This method is also used in `text_input.dart` as a private method. Is there anyway we can share that logic? (making it public. Or maybe add an `extension` on the `TextAffinity` enum to include this `_toTextAfinnity` function. 

```dart
extension TextAfinnityExtension on TextAffinity {
    TextAffinity? toTextAffinity(String? affinity) {
     .......
    }
}
```

Reply:
Another suggestion was to move the function to `text_editing.dart` and make it global and public.

Reply:
It's not terrible duplicated either, since it's so simple.  The `extension` idea might be out because the [style guide says](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#tldr) to avoid it.  Otherwise anything is probably fine, but I'm interested if anyone else has a strong opinion.